src/flux/scores.coffee

```diff
@@ -10,105 +11,111 @@ REJECTING = 'rejecting'
 REJECTED = 'rejected'
 
 
-
 allStudents = (scores) ->
   _.chain(scores)
     .pluck('students')
     .flatten(true)
     .value()
 
+getTaskInfoById = (taskId, data) ->
+  taskId = parseInt(taskId, 10)
+  for courseId, period of data
+    for period, periodIndex in data[courseId]
+      for student, studentIndex in period.students
+        for task in student.data
+          if task.id is taskId
+            return {task, courseId, period, periodIndex, studentIndex}
```

Comment:
Yeah, I thought about doing that, but didn't do so be cause that'd require setting up the cache overtime the performance report loads, and it probably won't actually be used too often, since the task lookup only happens if the teacher accepts late work.

However - your comment's made me think on it a bit more and what we actually want is a cache that is initialized on first access, then later accepts will be fast without impacting the first-display time.   I'll do that